akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -114,6 +115,22 @@ object Sink {
     new Sink(scaladsl.Sink.actorRef[In](ref, onCompleteMessage))
 
   /**
+   * Sends the elements of the stream to the given `ActorRef` that sends back back-pressure signal.
+   * First element is always `onInitMessage`, then stream is waiting for acknowledgement message
+   * `ackMessage` from the given actor which means that it is ready to process
+   * elements. It also requires `ackMessage` message after each stream element
+   * to make backpressure works.
+   *
+   * If the target actor terminates the stream will be canceled.
+   * When the stream is completed successfully the given `onCompleteMessage`
+   * will be sent to the destination actor.
+   * When the stream is completed with failure a [[akka.actor.Status.Failure]]
```

Comment:
perhaps make this configurable, too? could be `onFailureMessage: (Throwable) => Any = Status.Failure`